File: Olympiads/USACO/USACO 2026 First Contest, Bronze Problem 1. Chip Exchange.py
from math import ceil

for _ in range(int(input())):
    a,b,c,d,f = map(int,input().split(" "))


    a += c * (b // d)
    b %= d

    if(a >= f):
        print(0)
        continue

    ans = d - b - 1

    print(ans)

    opt1 = f - a


    opt2 = d - (f-a) % c

    a += c

    print(f-a, opt2)

    
    opt2 += ceil((f-a) / c) * d

    print(opt1, opt2, ans)

    print(ans + max(opt1, opt2))

File: Olympiads/USACO/USACO 2026 First Contest, Bronze Problem 1. Chip Exchange.cpp
#include <bits/stdc++.h>
#include <iostream>

using namespace std;

#define int long long

int32_t main(){
    int tt; cin >> tt;

    while(tt--){
        int a,b, ca,cb, fa; cin >> a >> b >> ca >> cb >> fa;

        a += ca * (b / cb);
        b %= cb;

        if(a >= fa){
            cout << "0\n";
            continue;
        }

        int ans = max(cb - b - 1, 0LL);


        int opt1 = max(fa - a, 0LL);

        //increase a so that fa - a = b-1 (mod b)
        cout <<fa- a << "\n";
        int opt2 = cb - ((fa-a) % ca);

        a += ca;

        cout << a << "\n";
        cout << (fa-a) << "\n";
        cout << ceil((fa-a)/(long double)ca) << " " << ceil((fa-a)/(long double)ca) * cb << "\n";

        opt2 += ceil((fa-a)/(long double)ca) * cb;

        cout << opt2 << "\n";

        cout << max(ans + max(opt1, opt2), 0LL) << "\n";
    
    }
}